core/plugins/org.polarsys.capella.core.platform.sirius.sirius.validation/src/org/polarsys/capella/core/platform/sirius/sirius/validation/ddiagram/CapellaElementInDescriptionExistanceCheck.java

```diff
@@ -12,68 +12,69 @@
  *******************************************************************************/
 package org.polarsys.capella.core.platform.sirius.sirius.validation.ddiagram;
 
-import java.io.StringReader;
-
-import javax.xml.XMLConstants;
-import javax.xml.parsers.SAXParser;
-import javax.xml.parsers.SAXParserFactory;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
 
 import org.apache.log4j.Logger;
 import org.eclipse.core.runtime.IStatus;
 import org.eclipse.emf.ecore.EObject;
 import org.eclipse.emf.validation.IValidationContext;
+import org.eclipse.emf.validation.model.ConstraintStatus;
+import org.eclipse.sirius.business.api.dialect.DialectManager;
+import org.eclipse.sirius.business.api.session.Session;
+import org.eclipse.sirius.viewpoint.DRepresentationDescriptor;
 import org.polarsys.capella.common.tools.report.config.registry.ReportManagerRegistry;
 import org.polarsys.capella.common.tools.report.util.IReportManagerDefaultComponents;
 import org.polarsys.capella.core.data.capellacore.CapellaElement;
-import org.polarsys.capella.core.model.utils.saxparser.IConstantValidation;
-import org.polarsys.capella.core.model.utils.saxparser.SaxParserHelper;
+import org.polarsys.capella.core.model.utils.NamingHelper;
 import org.polarsys.capella.core.validation.rule.AbstractValidationRule;
-import org.xml.sax.Attributes;
-import org.xml.sax.InputSource;
-import org.xml.sax.SAXException;
-import org.xml.sax.SAXParseException;
-import org.xml.sax.helpers.DefaultHandler;
 
 /**
  * Check if HyperLinks to capella element or diagram does not exist in model any more
  */
 public class CapellaElementInDescriptionExistanceCheck extends AbstractValidationRule {
 
-  public final class LocalDefaultHandler extends DefaultHandler {
-    private final CapellaElement capellaElement;
-    private final IStatus[] result;
-    private final IValidationContext ctx;
+  public List<IStatus> validateElement(EObject element, String description, IValidationContext ctx) {
+    List<IStatus> result = new ArrayList<>();
+    ILinkParserHandler linkParserHandler = new ILinkParserHandler() {
```

Comment:
I would name it ILinkParser only, not to confound with the DescriptionLinkParserHandler that will be used just after this. Same remark goes for the xxxModifierHandler